VerilogCodingStyle.md

```diff
@@ -797,6 +797,92 @@ endmodule : foo
 localparam bit ValBaz = 1;  // This comment describes the item to the left.
 ```
 
+It can sometimes be useful to structure the code using header-style comments in
+order to separate different functional parts (like FSMs, the main datapath or
+registers) within a module. In that case, the preferred style is as follows:
+
+```systemverilog
+module foo;
+
+  ////////////////
+  // Controller //
+  ////////////////
+  ...
+
+  ///////////////////////
+  // Main ALU Datapath //
+  ///////////////////////
+  ...
+
+  ///////////////////////////////////////////////////////
+  // Complicated part of the design that warrants some //
+  // more description in the header comment            //
+  ///////////////////////////////////////////////////////
+  ...
+
+  ///////////////////////////////////////////////////
+  // It is ok to omit the slashes on the right side
+  ///////////////////////////////////////////////////
+  ...
+
+  case (...)
+    /////////////////////////////////////////
+    // This is ok as well, e.g. in order to
+    // describe what an FSM state does in a
+    // switch-case statement
+    SomeState: begin
+    ...
+    end
+    ...
+  endcase
+
+endmodule : foo
+```
+
+If the designer would like to use comments to mark the beginning/end of a
+particular section for better readability (e.g. in nested for loop blocks), the
+preferred way is to use a single-line comment with no extra delineators, as
+shown in the examples below.
+
+&#x1f44d;
+```systemverilog
```

Comment:
@msfschaffner can you append ` {.good}` to this? it renders green on some viewers (though not yet on github). See other sections of this document for example